src/tablestore.rs

```diff
@@ -210,6 +212,80 @@ impl TableStore {
         }
     }
 
+    #[allow(dead_code)]
+    pub(crate) async fn delete_sst(&self, id: &SsTableId) -> Result<(), SlateDBError> {
+        let path = self.path(id);
+        self.object_store
+            .delete(&path)
+            .await
+            .map_err(SlateDBError::ObjectStoreError)
+    }
+
+    pub(crate) async fn collect_garbage_wal(
+        &self,
+        min_age: Duration,
+        state: &CoreDbState,
+    ) -> Result<(), SlateDBError> {
+        let min_age = chrono::Duration::from_std(min_age).expect("invalid duration");
+        let wal_list = self
+            // Delete all SSTs up to (but not including) the last compacted WAL SST
+            .get_wal_sst_list(..state.last_compacted_wal_sst_id)
```

Comment:
Similar to the comment below, I don't think we should have tablestore be aware of the wal compaction protocol. It seems cleaner to have the gc figure out what SSTs to retain, and pass that list into this method.